### PR TITLE
Windows: Native autogen

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard <jhoward@microsoft.com>

In order to get to full nanoserver support, I need to move away from bash scripts and onto native Windows scripts. This PR includes a re-worked version of the autogen script in PowerShell, and performs effectively what I do in my normal day-to-day workflow when building/developing docker on Windows. I'm just submitting it into the repo for others (including myself!) to make use of.

### CODE REVIEW

**Comment by jhowardmsft** [comment]
@friism @johnstep FYI

**Comment by jhowardmsft** [comment]
@thaJeztah This is one which would be very useful to take into 1.13. The reason being that once merged, I am going to be updating Windows CI to use this. Without this taken back into 1.13 RC, CI on the 1.13 branches would otherwise fail. 

(Obviously ultimately all the Windows CI files will be in the repo, but that isn't the case yet, and they're still a moving target while I'm working on supporting nanoserver and other scenarios post RS1).

File: hack/make/.go-autogen.ps1 [hunk 1]
```diff
@@ -0,0 +1,96 @@
+﻿<#
+.NOTES
+    Author:  @jhowardmsft
+
+    Summary: Windows native version of .go-autogen which generates the
+             .go source code for building, and performs resource compilation.
+
+.PARAMETER CommitString
+     The commit string. This is calculated externally to this script
```
Comment (jhowardmsft) [inline hack/make/.go-autogen.ps1]: Fixed

**Review by thaJeztah — APPROVED** [review]
I see @johnstep's nit was addressed, so let me LGTM

ping @friism @StefanScherer could you have a look at this one?

**Comment by StefanScherer** [comment]
Haven't tested it, but LGTM. So there will be an equivalent for `hack/make.sh` as well?

A lot things changed since the first idea of a [hack/make.ps1](https://github.com/hypriot/docker/blob/add-dockerfile-for-windows-withmake-ps1/hack/make.ps1#L22) 😄

**Comment by jhowardmsft** [comment]
Yes. I have mostly working make.ps1 now, integrated into CI. PR hopefully next week.

**Comment by jhowardmsft** [comment]
@vdemeester Any chance of reviewing this? I need this in place for the next piece of the puzzle. Thanks 😸

File: hack/make/.go-autogen.ps1 [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    Copy-Item "hack\make\.resources-windows\resources.go" "autogen\winresources\dockerd"
+
+    # Generate a version in the form major,minor,patch,build
+    $versionQuad=$DockerVersion -replace "[^0-9.]*" -replace "\.", ","
+
+    # Compile the messages
+    windmc hack\make\.resources-windows\event_messages.mc -h autogen\winresources\tmp -r autogen\winresources\tmp
+    if ($LASTEXITCODE -ne 0) { Throw "Failed to compile event message resources" }
+
+    # If you really want to understand this madness below, search the Internet for powershell variables after verbatim arguments... Needed to get double-quotes passed through to the compiler options.
```
Comment (jhowardmsft) [inline hack/make/.go-autogen.ps1]: No. Just tried.

Comment (johnstep) [inline hack/make/.go-autogen.ps1]: Okay, thanks.

**Review by johnstep — APPROVED** [review]

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**